webpack/assets/javascripts/react_app/routes/RegistrationCommands/RegistrationCommandsPage/components/fields/RepositoryModal.js

```diff
@@ -0,0 +1,180 @@
+      gpgKeyUrl: '',
+    },
+  ];
+
+  useEffect(() => {
+    if (!repoData.length) {
+      handleRepoData(initialRepoState);
+    }
+  }, [repoData, initialRepoState, handleRepoData]);
+
+  const [repoReferenceCounter, setRepoReferenceCounter] = useState(0);
+
+  const updateInput = (reference, input, inputType) => {
+    handleRepoData(current =>
+      current.map(repo => {
+        if (repo.reference === reference) {
+          if (inputType === 'repository') {
+            return { ...repo, repository: input };
+          }
+
+          return { ...repo, gpgKeyUrl: input };
+        }
+        return repo;
+      })
+    );
+  };
+
+  const removeRepository = reference => {
+    handleRepoData(current =>
+      current.filter(repo => repo.reference !== reference)
+    );
+  };
+
+  const addRepositoryModalInput = () => {
+    const newRepoReferenceCounter = repoReferenceCounter + 1;
+    handleRepoData(current => [
+      ...current,
+      {
+        reference: newRepoReferenceCounter,
+        repository: '',
+        gpgKeyUrl: '',
+      },
+    ]);
+    setRepoReferenceCounter(newRepoReferenceCounter);
+  };
+
+  const clearRepositories = () => {
+    handleRepoData(initialRepoState);
+    handleModalToggle();
+  };
+
+  const renderRepo = repo => (
+    <Split hasGutter key={'repo_modal_list'.concat(repo.reference.toString())}>
+      <SplitItem span={10}>
+        <RepositoryModalInput
+          reference={repo.reference}
+          updateRepository={updateInput}
+          isLoading={isLoading}
+          initRepo={repo.repository}
+          initGpgKey={repo.gpgKeyUrl}
+        />
+      </SplitItem>
+      <SplitItem>
+        <Button
+          ouiaId={repo.reference}
+          variant="link"
+          icon=<MinusCircleIcon />
+          onClick={() => removeRepository(repo.reference)}
+        >
+          {__('Remove')}
+        </Button>
+      </SplitItem>
+    </Split>
+  );
+
+  return (
+    <Modal
+      ouiaId="host_reg_repo_modal"
+      key="host_reg_repo_modal"
+      title={__('Repository list')}
+      isOpen={isModalOpen}
+      onClose={clearRepositories}
+      actions={[
+        <Button
+          ouiaId="reg_modal_confirm"
+          key="reg_modal_confirm"
+          variant="primary"
+          onClick={handleModalToggle}
+        >
+          {__('Confirm')}
+        </Button>,
+        <Button
+          ouiaId="reg_modal_cancel"
+          key="reg_modal_cancel"
+          variant="link"
+          onClick={clearRepositories}
+        >
+          {__('Cancel')}
+        </Button>,
+        <Button
+          ouiaId="reg_modal_add_new_repo"
+          key="reg_modal_add_new_repo"
+          variant="link"
+          icon={<PlusCircleIcon />}
+          onClick={addRepositoryModalInput}
+        >
+          {__('Add repository')}
+        </Button>,
+      ]}
+    >
+      <Split hasGutter>
+        <Grid hasGutter md={6}>
+          <FormGroup
+            label={__('Repository')}
```

Comment:
@MariaAga I had a hard time aligning the labels on the Modal and the text input fields. The corresponding input fields for repos/gpg keys should be aligned with the heading above. Also, I want to place the remove button for one line next to the input fields to the right. Could you give me a hint how to do this best, please?

Reply:
Could you attach a simple sketch of what you mean? I see the model as you describe you want it to be
![image](https://github.com/theforeman/foreman/assets/30431079/c51dcebd-327b-40e7-8bc6-58917b5b8502)

Reply:
In your screen shot it looks like it should. For me, it looks like this: 
![image](https://github.com/theforeman/foreman/assets/71487468/d2a71406-35b7-4f5e-93db-bbc8905e3005)
The headings are not aligned with the input fields.

Reply:
Ok disabling  mark_translated now shows me what you see, I'll take a look

Reply:
![image](https://github.com/theforeman/foreman/assets/30431079/49ae02e2-52b1-4474-a9d1-471617dbf164)
I removed the Split and created a grid which is like
```
<Grid> 
<Item> repo label</item> <Item> GPG label</item> <item>empty space</item>
<Item> input</item> <Item>input</item> <item>remove</item>
</Grid>
```
And specifically: 
https://gist.github.com/MariaAga/0efbad6993ecf112d2d29bab71cf0e4e

Reply:
and if the model shouldn't grow too much with the screen you can use `<Model variant={ModalVariant.small}`

Reply:
Thanks a lot @MariaAga , it looks exactly as I wanted and I finally understood how to make the Grid work!